### ISSUE TITLE
Unable to determine if superagent is dismissing the body in response of a request

### ISSUE THREAD

**Issue by silverlight513** [forwardemail/superagent#1160]
I've just been dealing with an API that set the content type header to `application-json` instead of `application/json` as it should be. This meant that superagent set the res.body to the default of `{}`. This is correct as `application-json` is an incorrect value. 

The problem comes when you test the request in the browser (in my case, Chrome v55). The browser doesn't care too much about this slight change in string and parses the body as json. As no error is returned by either superagent or the browser it's difficult to see what the problem is. It took us a day to find the issue and have now changed the API.

I think that superagent should default to an empty object but show a warning of sorts explaining to the user that the body in the superagent response has been defaulted to an empty object as the content-type doesn't match a known string.

Is a warning message something that can be added in for this case or would this not be a valid request?

**Comment by pornel** [comment]
Yes, it's unfortunate that our default for `res.body` is `{}`. It should probably be `undefined`, but it's a breaking change :(

**Comment by titanism** [comment]
application-json is malformed, so automatic JSON parsing is not selected and raw text remains available. Changing legacy default/warnings requires a separate compatibility design; correct the server header to application/json.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.

**Comment by titanism** [comment]
Closing this for now. If this still affects v10.4.0, please open a new issue with a current reproduction.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.
